File: dev-c++/0231-power-of-two.cpp
class Solution {
public:
    bool isPowerOfTwo(int n) {
        if(n == 0) { return false; }
        if(n == 1) { return true;  }

        long mask = 2;

        for(int i = 0; i < 31; ++ i) {
            if(n == mask) { return true; }
            mask <<= 1;
        }

        return false;
    }
};

File: dev-rust/0231-power-of-two.rs
impl Solution {
    pub fn is_power_of_two(n: i32) -> bool {
        if n == 0 {
            return false;
        } 

        if n == 1 {
            return true;
        }

        let mut mask: i64 = 2;
        let n_64: i64 = n.into();

        while mask < (i32::MAX as i64) {
            if n_64 == mask {
                return true;
            }

            mask <<= 1;
        }

        return false;
    }
}
